hypothesis-python/src/hypothesis/internal/conjecture/choicetree.py

```diff
@@ -9,16 +9,19 @@
 # obtain one at https://mozilla.org/MPL/2.0/.
 
 from collections import defaultdict
+from typing import Any, Callable, Dict, Iterable, Iterator, List, Optional, Sequence
 
 from hypothesis.internal.conjecture.junkdrawer import LazySequenceCopy, pop_random
 
 
-def prefix_selection_order(prefix):
+def prefix_selection_order(
+    prefix: Sequence[int],
+) -> Callable[[int, int], Iterator[int]]:
```

Comment:
In general I think we always want to use `Iterable` over `Iterator`; so long as we only ever need the more general interface it's nice to have that consistency.